galasa-simbank-tests/dev.galasa.simbank.manager/src/main/java/dev/galasa/simbank/manager/internal/gherkin/SimbankStatementOwner.java

```diff
@@ -0,0 +1,117 @@
+import dev.galasa.framework.spi.language.gherkin.ExecutionMethod;
+import dev.galasa.framework.spi.language.gherkin.GherkinKeyword;
+import dev.galasa.http.HttpClientException;
+import dev.galasa.http.IHttpClient;
+import dev.galasa.simbank.manager.IAccount;
+import dev.galasa.simbank.manager.SimBankManagerException;
+import dev.galasa.simbank.manager.internal.SimBankImpl;
+import dev.galasa.simbank.manager.internal.SimBankManagerImpl;
+
+public class SimbankStatementOwner implements IStatementOwner {
+
+    private final static String webAPISoap = "<soapenv:Envelope xmlns:soapenv='http://schemas.xmlsoap.org/soap/envelope/'>"
+            + "<soapenv:Body>" + "<ns1:UPDACCTOperation xmlns:ns1='http://www.UPDACCT.STCUSTN2.Request.com'>"
+            + "<ns1:update_account_record>" + "<ns1:account_key>" + "<ns1:sort_code>00-00-00</ns1:sort_code>"
+            + "<ns1:account_number>++ACCOUNT_NUMBER++</ns1:account_number>" + "</ns1:account_key>"
+            + "<ns1:account_change>++AMOUNT++</ns1:account_change>"
+            + "</ns1:update_account_record></ns1:UPDACCTOperation>" + "</soapenv:Body>" + "</soapenv:Envelope>\n";
+
+    SimBankManagerImpl manager;
+    IHttpClient client;
+    IAccount provisionedAccount;
+
+    public SimbankStatementOwner(SimBankManagerImpl manager) {
+        this.manager = manager;
+    }
+
+    public void setHttpClient(IHttpClient client) {
+        this.client = client;
+    }
+
+    @ExecutionMethod(keyword = GherkinKeyword.GIVEN, regex = "The Simbank is available")
+    public void generateSimBank(IGherkinExecutable executable, Map<String, Object> testVariables)
+            throws SimBankManagerException {
+        manager.generateSimBank(true, true);
+    }
+
+    @ExecutionMethod(keyword = GherkinKeyword.WHEN, regex = "I navigate to SimBank")
+    public void navigateToMainMenu(IGherkinExecutable executable, Map<String, Object> testVariables)
+            throws SimBankManagerException {
+        SimBankImpl bank = manager.getSimBank();
+        bank.start();
+    }
+
+    @ExecutionMethod(keyword = GherkinKeyword.THEN, regex = "I should see the main screen")
+    public void onMainMenu(IGherkinExecutable executable, Map<String, Object> testVariables)
+            throws SimBankManagerException {
+        SimBankImpl bank = manager.getSimBank();
+        String screen = bank.getControlTerminal().retrieveScreen();
+
+        assertOnScreen(screen, "Options     Description        PFKey ");
+        assertOnScreen(screen, "BROWSE      Browse Accounts    PF1");
+        assertOnScreen(screen, "TRANSF      Transfer Money     PF4");
+    }
+
+    public void assertOnScreen(String screen, String expected) throws SimBankManagerException {
+        if (!screen.contains(expected)) {
+            throw new SimBankManagerException("Could not find:\n" + expected + "\n\nScreen:\n" + screen);
+        }
+    }
+
+    @ExecutionMethod(keyword = GherkinKeyword.GIVEN, regex = "I have an account with a balance of ([\\d.]+)")
```

Comment:
[\\d.]+ can result in an invalid decimal number,   12.45.56
(\d+\.\d\d) would be safer